application/whatsapp/v0/send_message.go

```diff
@@ -0,0 +1,586 @@
+type TaskSendMessageOutput struct {
+	WaID string `json:"recipient-wa-id"`
+	ID   string `json:"message-id"`
+}
+
+// ----------------------- Tasks -----------------------
+
+// Task 1: Send Text Message Task
+
+type TaskSendTextMessageInput struct {
+	PhoneNumberID string `json:"phone-number-id"`
+	To            string `json:"to"`
+	Body          string `json:"body"`
+	PreviewURL    string `json:"preview-url"`
+}
+
+type TaskSendTextMessageReq struct {
+	MessagingProduct string     `json:"messaging_product"`
+	To               string     `json:"to"`
+	Type             string     `json:"type"`
+	Text             textObject `json:"text"`
+}
+
+func (e *execution) SendTextMessage(in *structpb.Struct) (*structpb.Struct, error) {
+	inputStruct := TaskSendTextMessageInput{}
+	err := base.ConvertFromStructpb(in, &inputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert input to struct: %v", err)
+	}
+
+	req := TaskSendTextMessageReq{
+		MessagingProduct: "whatsapp",
+		To:               inputStruct.To,
+		Type:             "text",
+		Text: textObject{
+			Body:       inputStruct.Body,
+			PreviewURL: inputStruct.PreviewURL == "true",
+		},
+	}
+
+	resp, err := e.client.SendMessageAPI(&req, &TaskSendMessageResp{}, inputStruct.PhoneNumberID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	respStruct := resp.(*TaskSendMessageResp)
+
+	outputStruct := TaskSendMessageOutput{
+		WaID: respStruct.Contacts[0].WaID,
+		ID:   respStruct.Messages[0].ID,
+	}
+
+	output, err := base.ConvertToStructpb(outputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert output to struct: %v", err)
+	}
+
+	return output, nil
+}
+
+// Task 2: Send Media Message Task
+
+type TaskSendMediaMessageInput struct {
+	PhoneNumberID string `json:"phone-number-id"`
+	To            string `json:"to"`
+	MediaType     string `json:"media-type"`
+	IDOrLink      string `json:"id-or-link"`
+	Caption       string `json:"caption"`  //cannot be used in audio
+	Filename      string `json:"filename"` //only for document
+
+}
+
+type TaskSendMediaMessageReq struct {
+	MessagingProduct string       `json:"messaging_product"`
+	To               string       `json:"to"`
+	Type             string       `json:"type"`
+	Document         *mediaObject `json:"document,omitempty,"`
+	Audio            *mediaObject `json:"audio,omitempty"`
+	Image            *mediaObject `json:"image,omitempty"`
+	Video            *mediaObject `json:"video,omitempty"`
+}
+
+func (e *execution) TaskSendMediaMessage(in *structpb.Struct) (*structpb.Struct, error) {
+	inputStruct := TaskSendMediaMessageInput{}
+	err := base.ConvertFromStructpb(in, &inputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert input to struct: %v", err)
+	}
+
+	req := TaskSendMediaMessageReq{
+		MessagingProduct: "whatsapp",
+		To:               inputStruct.To,
+	}
+
+	var id string
+	var link string
+	if strings.Contains(inputStruct.IDOrLink, "http") {
```

Comment:
important

not sure what ID will be. But, I think it will be accurate that we use prefix to judge.
[ref](https://stackoverflow.com/questions/12667327/go-startswithstr-string)